src/movies.js

```diff
@@ -1,22 +1,71 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+
+function getAllDirectors(moviesArray) {
+    const directors = moviesArray.map((movies) => movies.director);
+    return directors;
+}
+
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(array) {
+    const movieTypes = array.filter((movies) => movies.director === "Steven Spielberg" && movies.genre.includes("Drama"));
+    
+    return movieTypes.length;
+ }// Iteration 3: All scores average - Get the average of all scores with 2 decimals
+function scoresAverage(scoresArray) {
+    if ( scoresArray.length === 0) {
```

Comment:
There's nothing wrong here, however, the reduce function won't execute if the moviesArray is empty. So, the code block `if (scoresArray.length === 0) return 0;` is not needed as the function will work fine without it, it will return 0 as default value for empty array.